src/datetime.rs

```diff
@@ -1,12 +1,147 @@
 use std::{
-    convert::TryInto,
+    convert::{TryFrom, TryInto},
     error,
     fmt::{self, Display},
     result,
     time::{Duration, SystemTime},
 };
 
-use time::format_description::well_known::Rfc3339;
+use builder::*;
+use time::{format_description::well_known::Rfc3339, Date};
+
+mod builder {
```

Comment:
we generally try (with the exception of tests sometimes) to have a single module per file. So with that, could we split this out to a `datetime/builder.rs` file?